src/routes/relay/entities/schema/sponsored-call.schema.helper.ts

```diff
@@ -19,6 +20,45 @@ const isCalldata = (data: string, signature: string): boolean => {
   return data.startsWith(signatureId);
 };
 
+/**
+ * Checks whether data is a call to any method in the specified singleton
+ * @param singletonDeployment singleton deployment
+ * @param data call data
+ * @returns boolean
+ */
+const isSingletonCalldata = (
+  singletonDeployment: SingletonDeployment,
+  data: string,
+): boolean => {
+  const deploymentInterface = new ethers.Interface(singletonDeployment.abi);
+
+  return deploymentInterface.fragments.some((fragment) => {
+    if (ethers.FunctionFragment.isFunction(fragment)) {
+      const signature = fragment.format();
+      return isCalldata(data, signature);
+    }
+  });
+};
+
+/**
+ * Checks whether data is a call to any method in the Safe singleton
+ * @param data call data
+ * @returns boolean
+ */
+export const isSafeCalldata = (data: string): boolean => {
+  const safeL1Deployment = getSafeSingletonDeployment();
```

Comment:
Should we consider the Safe version?
Some functions like `changeMasterCopy` were removed in 1.3.0. 
So people could not update their Safes on Gnosis Chain (as we removed the option to execute with owner)

Reply:
Good point. However, I think the added overhead of checking _all_ contract ABIs isn't worth it considering [we intend to allow execution with the owner wallet](https://github.com/safe-global/web-core/issues/1894) again. What do you think?

Reply:
True. Let's stick with the latest deployment then :)